src/main/java/bio/terra/cli/service/WorkspaceManagerServiceGcp.java

```diff
@@ -331,6 +339,51 @@ public GcpAiNotebookInstanceResource createControlledGcpNotebookInstance(
         "Error creating controlled GCP Notebook instance in the workspace.");
   }
 
+  /**
+   * Call the Workspace Manager POST
+   * "/api/workspaces/v1/{workspaceId}/resources/controlled/gcp/dataproc-clusters" endpoint to add a
+   * GCP dataproc cluster as a controlled resource in the workspace.
+   *
+   * @param workspaceId the workspace to add the resource to
+   * @param createParams resource definition to create
+   * @return the GCP dataproc cluster resource object
+   */
+  public GcpDataprocClusterResource createControlledGcpDataprocCluster(
+      UUID workspaceId, CreateGcpDataprocClusterParams createParams) {
+    // convert the CLI object to a WSM request object
+    String jobId = UUID.randomUUID().toString();
+    CreateControlledGcpDataprocClusterRequestBody createRequest =
+        new CreateControlledGcpDataprocClusterRequestBody()
+            .common(createCommonFields(createParams.resourceFields))
+            .dataprocCluster(toWsmCreateClusterParams(createParams))
+            .jobControl(new JobControl().id(jobId));
+    logger.debug("Create controlled GCP Dataproc cluster request {}", createRequest);
+
+    return handleClientExceptions(
+        () -> {
+          ControlledGcpResourceApi controlledGcpResourceApi =
+              new ControlledGcpResourceApi(apiClient);
+          // Start the GCP Dataproc cluster creation job.
+          HttpUtils.callWithRetries(
+              () -> controlledGcpResourceApi.createDataprocCluster(createRequest, workspaceId),
+              WorkspaceManagerService::isRetryable);
+
+          // Poll the result endpoint until the job is no longer RUNNING.
+          CreatedControlledGcpDataprocClusterResult createResult =
+              HttpUtils.pollWithRetries(
+                  () -> controlledGcpResourceApi.getCreateDataprocClusterResult(workspaceId, jobId),
+                  (result) -> isDone(result.getJobReport()),
+                  WorkspaceManagerService::isRetryable,
+                  // Creating a GCP Dataproc cluster should take less than ~15 minutes.
```

Comment:
15 mins is a considerable time. Can we return immediately and expect the user to check status later?
same approach is used in AWS notebooks (which can table 5-10 mins as well usually)